connector/routingconnector/traces_test.go

```diff
@@ -204,6 +204,135 @@ func TestTracesCorrectlySplitPerResourceAttributeWithOTTL(t *testing.T) {
 	})
 }
 
+func TestTracesCorrectlyMatchOnceWithOTTL(t *testing.T) {
+	tracesDefault := component.NewIDWithName(component.DataTypeTraces, "default")
+	traces0 := component.NewIDWithName(component.DataTypeTraces, "0")
+	traces1 := component.NewIDWithName(component.DataTypeTraces, "1")
+
+	cfg := &Config{
+		DefaultPipelines: []component.ID{tracesDefault},
+		MatchOnce:        true,
+		Table: []RoutingTableItem{
+			{
+				Statement: `route() where attributes["value"] > 0 and attributes["value"] < 4`,
+				Pipelines: []component.ID{traces0},
+			},
+			{
+				Statement: `route() where attributes["value"] > 1 and attributes["value"] < 4`,
```

Comment:
Good test cases, thanks for adding! Another might be where it doesn't match the first, but does successfully match the second statement and gets routed there. If we update the values slightly it should be pretty simple to add another test.

Same comment goes for logs and metrics tests. We need to confirm that telemetry can be routed to the non-default and not-first `table` defined in case there's some kind of edge case we're missing.

Reply:
> Good test cases, thanks for adding! Another might be where it doesn't match the first, but does successfully match the second statement and gets routed there. If we update the values slightly it should be pretty simple to add another test.

I think the test case "span matched by one expression, multiple pipelines" meets your requirements. It sets attributes["value"] to 5, which causes it to not match the first two rules but match the third rule "route() where attributes["value"] == 5". As a result, it sends the data to both tracesDefault and traces0.

Reply:
Good point. I didn't put it together that it was sending to both `tracesDefault` and `traces0`, which will only happen in the case where that last condition is successfully met. 👍